Content.Server/_White/Headcrab/HeadcrabSystem.cs

```diff
@@ -0,0 +1,202 @@
+
+        _popup.PopupEntity(Loc.GetString("headcrab-hit-entity-head",
+                ("entity", args.Equipee)),
+            uid, uid, PopupType.LargeCaution);
+
+        _popup.PopupEntity(Loc.GetString("headcrab-eat-other-entity-face",
+            ("entity", args.Equipee)), args.Equipee, Filter.PvsExcept(uid), true, PopupType.Large);
+
+        _stunSystem.TryParalyze(args.Equipee, TimeSpan.FromSeconds(component.ParalyzeTime), true);
+        _damageableSystem.TryChangeDamage(args.Equipee, component.Damage, origin: uid);
+    }
+
+    private void OnUnequipAttempt(EntityUid uid, HeadcrabComponent component, BeingUnequippedAttemptEvent args)
+    {
+        if (args.Slot != "mask" ||
+            component.EquippedOn != args.Unequipee ||
+            HasComp<ZombieComponent>(args.Unequipee) ||
+            _mobState.IsDead(uid))
+            return;
+
+        _popup.PopupEntity(Loc.GetString("headcrab-try-unequip"),
+            args.Unequipee, args.Unequipee, PopupType.Large);
+        args.Cancel();
+    }
+
+    private void OnGotEquippedHand(EntityUid uid, HeadcrabComponent component, GotEquippedHandEvent args)
+    {
+        if (_mobState.IsDead(uid) ||
+            HasComp<ZombieComponent>(args.User) ||
+            HasComp<GhostComponent>(args.User))
+            return;
+
+        _handsSystem.TryDrop(args.User, uid, checkActionBlocker: false);
+        _damageableSystem.TryChangeDamage(args.User, component.Damage);
+        _popup.PopupEntity(Loc.GetString("headcrab-entity-bite"),
+            args.User, args.User);
+    }
+
+    private void OnGotUnequipped(EntityUid uid, HeadcrabComponent component, GotUnequippedEvent args)
+    {
+        if (args.Slot != "mask")
+            return;
+
+        component.EquippedOn = EntityUid.Invalid;
+        RemCompDeferred<PacifiedComponent>(uid);
+        var combatMode = EnsureComp<CombatModeComponent>(uid);
+        _combat.SetInCombatMode(uid, true, combatMode);
+        EnsureComp<NPCMeleeCombatComponent>(uid);
+        _npcFaction.RemoveFaction(args.Equipee, "Zombie");
+    }
+
+    private void OnMeleeHit(EntityUid uid, HeadcrabComponent component, MeleeHitEvent args)
+    {
+        if (!args.HitEntities.Any()
+            || _random.Next(1, 101) <= component.ChancePounce)
+            return;
+
+        TryEquipHeadcrab(uid, args.HitEntities.First(), component);
+    }
+
+    private void OnJump(EntityUid uid, HeadcrabComponent component, JumpActionEvent args)
+    {
+        if (args.Handled || component.EquippedOn is { Valid: true })
+            return;
+
+        args.Handled = true;
+        var xform = Transform(uid);
+        var mapCoords = _transform.ToMapCoordinates(args.Target);
+        var direction = mapCoords.Position - _transform.GetMapCoordinates(xform).Position;
+
+        _throwing.TryThrow(uid, direction, 7F, uid, 10F);
+        if (component.JumpSound != null)
+            _audioSystem.PlayPvs(component.JumpSound, uid, component.JumpSound.Params);
```

Comment:
```suggestion
        _audioSystem.PlayPvs(component.JumpSound, uid, component.JumpSound.Params);
```
`PlayPvs` checks if sound is null so you don't have to/